infrastructure/workload-management.tf

```diff
@@ -23,7 +23,7 @@ module "synapse_management" {
   service_name        = local.service_name
 
   data_lake_account_id                   = module.synapse_data_lake.data_lake_account_id
-  deploy_purview                         = true
+  deploy_purview                         = var.create_purview_account

```

Comment:
Ensure this flag correctly handles scenarios with purview account limits or errors with quota.
You can also maybe log warnings when the Purview resource fails or is skipped

Reply:
That's what this boolean flag does. The account limit has already been reached since there are accounts in the dev, test and prod envs. This allows the account to be dropped in the build env